browser/net/brave_referrals_network_delegate_helper.cc

```diff
@@ -19,13 +21,16 @@ int OnBeforeStartTransaction_ReferralsWork(
     net::HttpRequestHeaders* headers,
     const ResponseCallback& next_callback,
     std::shared_ptr<BraveRequestInfo> ctx) {
-  if (!ctx->referral_headers_list)
-    return net::OK;
   // If the domain for this request matches one of our target domains,
   // set the associated custom headers.
+  PrefService* prefs = g_browser_process->local_state();
+  const base::ListValue* referral_headers_list =
+      prefs->GetList(kReferralHeaders);
```

Comment:
I was trying to explain that this is preference is also obsolete now. Since the list is permanent, there is no need to save the hardcoded value to prefs and then load, you can just filter URLs and add hardcoded headers in this function. Or do it in `BraveReferralsService::GetMatchingReferralHeaders`, but without juggling with the pref (that could actually be removed from both local state and profile prefs prefs now)

Reply:
i made it a singleton. fixed